app/views/PushTroubleshootView/index.tsx

```diff
@@ -0,0 +1,169 @@
+			// consumptionPercentage: state.troubleshootingNotification.consumptionPercentage,
+		})
+	);
+
+	const [testPushNotificationsPermission] = usePermissions(['test-push-notifications']);
+
+	useEffect(() => {
+		if (foreground) {
+			dispatch(requestTroubleshootingNotification());
+		}
+	}, [dispatch, foreground]);
+
+	useEffect(() => {
+		navigation.setOptions({
+			title: I18n.t('Push_Troubleshooting')
+		});
+	}, [navigation]);
+
+	const openNotificationDocumentation = async () => {
+		await Linking.openURL('https://docs.rocket.chat/use-rocket.chat/rocket.chat-mobile/push-notifications');
+	};
+
+	const alertDeviceNotificationSettings = () => {
+		showErrorAlert(
+			I18n.t('Device_notifications_alert_description'),
+			I18n.t('Device_notifications_alert_title'),
+			goToNotificationSettings
+		);
+	};
+
+	// TODO: This will be used in the near future when the consumption percentage is implemented on the server.
+	// const alertWorkspaceConsumption = () => {
+	// 	Alert.alert(I18n.t('Push_consumption_alert_title'), I18n.t('Push_consumption_alert_description'));
+	// };
+
+	const goToNotificationSettings = () => {
+		if (isIOS) {
+			Linking.openURL('app-settings:');
+		} else {
+			notifee.openNotificationSettings();
+		}
+	};
+
+	const handleTestPushNotification = async () => {
+		let message = '';
+		try {
+			const result = await Services.pushTest();
+			message = I18n.t('Your_push_was_sent_to_s_devices', { s: result.params[0] });
+		} catch (error: any) {
+			message = I18n.isTranslated(error?.error) ? I18n.t(error?.error) : error?.message;
+		} finally {
+			Alert.alert(I18n.t('Test_push_notification'), message);
+		}
+	};
+
+	let pushGatewayInfoDescription = 'Push_gateway_not_connected_description';
+	let pushGatewayStatusColor = colors.userPresenceBusy;
+	if (pushGatewayEnabled) {
+		pushGatewayStatusColor = colors.userPresenceOnline;
+		pushGatewayInfoDescription = 'Push_gateway_connected_description';
+	}
+	if (pushGatewayEnabled && !defaultPushGateway) {
+		pushGatewayStatusColor = colors.badgeBackgroundLevel3;
+		pushGatewayInfoDescription = 'Custom_push_gateway_connected_description';
+	}
+
+	return (
+		<SafeAreaView testID='push-troubleshoot-view'>
+			<StatusBar />
+			<List.Container testID='push-troubleshoot-view-list'>
+				<CustomListSection
+					title='Device_notification_settings'
+					statusColor={!deviceNotificationEnabled ? colors.userPresenceBusy : colors.userPresenceOnline}
+				>
+					<List.Separator />
+					<List.Item
+						title={!deviceNotificationEnabled ? 'Allow_push_notifications_for_rocket_chat' : 'Go_to_device_settings'}
+						onPress={!deviceNotificationEnabled ? alertDeviceNotificationSettings : undefined}
+						testID='push-troubleshoot-view-allow-push-notifications'
+					/>
+					<List.Separator />
+				</CustomListSection>
+
+				{/* TODO: This will be used in the near future when the consumption percentage is implemented on the server.
```

Comment:
Is `near future` really near?
Is push quote still a thing on CE?
Can you remove this and add to a new branch without a PR?
Link the branch to the PR and we can take it from there, if needed.
The code is very straightforward. It's not like we would waste time resolving a difficult conflict.
I can see there's more code comment on other files. Can you do it for all of them?

Reply:
https://github.com/RocketChat/Rocket.Chat.ReactNative/tree/feat.troubleshooting-notification-push-quota